Introduce bubble tea to cli framework

bjoginapally opened the pull request:
# Description

- Add bubble tea prompter to cli framework
- Change delete commands to use bubbletea framework


PS:  Follow up PR for init changes
## Issue reference

<!--
We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.
-->

Partially addresses: #3947 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Adds necessary unit tests for change
* [ ] Adds necessary E2E tests for change
* [x] Unit tests passing
* [ ] Extended the documentation / Created issue for it

<img width="978" alt="Screenshot 2023-01-05 at 10 30 52 AM" src="https://user-images.githubusercontent.com/99223262/210854415-54b9a4a0-9478-41a8-aec2-a67ec9cbd6e4.png">
<img width="984" alt="Screenshot 2023-01-05 at 10 31 09 AM" src="https://user-images.githubusercontent.com/99223262/210854417-cb044d52-2378-44d2-8783-d04065a623b8.png">

Commit:
Introduce bubble tea to cli framework
- Add bubble tea prompter to cli framework
- Change delete command to use bubbletea framework

issue: https://github.com/project-radius/radius/issues/3947

Commit:
Abstract out yes or no prompts

Commit:
Rename prompter interface

ytimocin reviewed (approved):
lgtm

[we should have a test for the change in prompt.go `fmt.Println`]

Commit:
Return boolean instead of text for yes or no inputs

Commit:
Change default yes or no option

Commit:
Correct tests

AaronCrawfis reviewed (approved):
UI/wording/experience looks good to me

Commit:
Remove explicit prompt on command line
-  This change passes the prompt message to bubble tea instead of explicit terminal printing
